home_work_7/src/com/bosiy/Book.java

```diff
@@ -0,0 +1,60 @@
+package com.bosiy;
+
+public class Book {
+    private String bookName;
+    private String writer;
+    private int yearOfPublishing;
+
+    public Book(String name) {
+        this.bookName = name;
+    }
+    public void allInfoFirstAttribute() {
+        System.out.println("Name of book: " + bookName + "\n");
+    }
+    public Book(String name, String writer) {
+        this.bookName = name;
+        this.writer = writer;
+    }
+    public void allInfoSecondAttribute() {
+        System.out.println("Name of book: " + bookName + "\nWriter: " + writer + "\n");
+    }
+    public Book(String name, String writer, int yearOfPublishing) {
+        if(yearOfPublishing <= 0) {
```

Comment:
0 is magic number, please move it to constants

also why aren't we able to create book with year of pub < 0? dont we have books which were written before

Reply:
fixed